src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -111,11 +136,20 @@ public virtual AnalyzerReference ReadAnalyzerReferenceFrom(ObjectReader reader,
         cancellationToken.ThrowIfCancellationRequested();
 
         var type = reader.ReadString();
-        if (type == nameof(AnalyzerFileReference))
+        switch (type)
         {
-            var fullPath = reader.ReadRequiredString();
-            var shadowCopy = reader.ReadBoolean();
-            return new AnalyzerFileReference(fullPath, _analyzerLoaderProvider.GetLoader(new AnalyzerAssemblyLoaderOptions(shadowCopy)));
+            case nameof(AnalyzerFileReference):
+                var fullPath = reader.ReadRequiredString();
+                var shadowCopy = reader.ReadBoolean();
+                return new AnalyzerFileReference(fullPath, _analyzerLoaderProvider.GetLoader(new AnalyzerAssemblyLoaderOptions(shadowCopy)));
+
+            case nameof(AnalyzerImageReference):
+                var guid = reader.ReadGuid();
+                lock (s_analyzerImageReferenceMapGate)
+                {
+                    Contract.ThrowIfFalse(s_analyzerImageReferenceMap.TryGetKey(guid, out var analyzerImageReference));
```

Comment:
>Contract.ThrowIfFalse(s_analyzerImageReferenceMap.TryGetKey(guid, out var analyzerImageReference)); [](http://example.com/codeflow?start=20&length=99)

nit: might be nice to have this be a method similar to what was done for TryGetValue

Reply:
will do!

Reply:
done in e035fc74062